packages/history/src/index.ts

```diff
@@ -51,17 +52,12 @@ export interface ParsedPath {
 
 export interface HistoryState {
   key?: string
+  index: number
```

Comment:
shouldn't this be `index?: number` as the index is not set for the initial entry?

also, i think we should use a more ... obscure name to not collide with user data

e.g. `__TSR_index` ?

Reply:
It's always available after we run through `parseHref`, this is because, if the route has no state (should be valid only for initial route), we default it to `{ index: 0 }`. About the naming, sure, but by that logic, shouldn't the `key` also be renamed?

Reply:
ideally, yes. but the `key` has been there since the beginning, but adding an `index` now could introduce a breaking change